src/FluffySpoon.AspNet.LetsEncrypt/CertificateRenewal.cs

```diff
@@ -0,0 +1,221 @@
+using System;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Certes;
+using Certes.Acme;
+using Certes.Acme.Resource;
+using FluffySpoon.AspNet.LetsEncrypt.Exceptions;
+using FluffySpoon.AspNet.LetsEncrypt.Persistence;
+using FluffySpoon.AspNet.LetsEncrypt.Persistence.Models;
+using Microsoft.Extensions.Logging;
+using static FluffySpoon.AspNet.LetsEncrypt.CertificateRenewalStatus;
+
+namespace FluffySpoon.AspNet.LetsEncrypt
+{
+	public class CertificateRenewal : ICertificateRenewal
+	{
+		public const string CertificateFriendlyName = "FluffySpoonAspNetLetsEncryptCertificate";
+
+		private readonly IPersistenceService _persistenceService;
+		private readonly ILogger<ICertificateRenewal> _logger;
+		private readonly LetsEncryptOptions _options;
+
+		private IAcmeContext _acme;
+
+		public CertificateRenewal(
+			IPersistenceService persistenceService,
+			ILogger<ICertificateRenewal> logger,
+			LetsEncryptOptions options)
+		{
+			_persistenceService = persistenceService;
+			_logger = logger;
+			_options = options;
+		}
+
+        public async Task<(X509Certificate2,CertificateRenewalStatus)> RenewCertificateIfNeeded(X509Certificate2 current)
```

Comment:
Instead of a Tuple here, can we use a struct or class? It's always more readable in my opinion.

Reply:
@dv00d00 did you see this?

Reply:
I've replaced it with `CertificateRenewalResult` class

Reply:
Ah didn't catch that - thanks.